File: javascript/leetcode/valid-anagram.js
/**
 *  * @param {string} s
 *   * @param {string} t
 *    * @return {boolean}
 *     */
var isAnagram = function(s, t) {
	var letCntsS = {}
	var letCntsT = {}
	if (s.length != t.length) { return false }
	for (var i = 0; i < s.length; i++) {
		letCntsS[s[i]] = letCntsS[s[i]] ? ++letCntsS[s[i]] : 1
	}
	for (var i = 0; i < t.length; i++) {
		letCntsT[t[i]] = letCntsT[t[i]] ? ++letCntsT[t[i]] : 1
		if (letCntsT[t[i]] > (letCntsS[t[i]] || 0)) { return false }
	}
	return true
};

File: python/leetcode/valid-anagram.py
class Solution:
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        letCntsS = {}
        letCntsT = {}
        if len(s) != len(t):
            return False
        for c in s:
            if c in letCntsS:
                letCntsS[c] = letCntsS[c] + 1
            else:
                letCntsS[c] = 1
        for c in t:
            if c in letCntsT:
                letCntsT[c] = letCntsT[c] + 1
            else:
                letCntsT[c] = 1
            if c not in letCntsS or letCntsT[c] > letCntsS[c]:
                return False
        return True
